pkg/oc/cli/cmd/tag.go

```diff
@@ -132,20 +132,16 @@ func parseStreamName(defaultNamespace, name string) (string, string, error) {
 	return namespace, streamName, nil
 }
 
-func determineSourceKind(f *clientcmd.Factory, input string) string {
-	mapper, _ := f.Object()
-	gvks, err := mapper.KindsFor(schema.GroupVersionResource{Group: imageapi.GroupName, Resource: input})
-	if err == nil {
-		return gvks[0].Kind
-	}
-
+func determineSourceKind(f *clientcmd.Factory, input string) (string, error) {
 	// DockerImage isn't in RESTMapper
 	switch strings.ToLower(input) {
 	case "docker", "dockerimage":
-		return "DockerImage"
+		return "DockerImage", nil
 	}
 
-	return input
+	mapper, _ := f.Object()
+	gvks, err := mapper.KindsFor(schema.GroupVersionResource{Group: imageapi.GroupName, Resource: input})
```

Comment:
Are you sure that when there's an error gvks will contain anything? I mean the next line might result in index out of bounds error if gvks is empty, nil. You should be still checking the error here.

Reply:
Oh, I understand. Thank you.